include/picongpu/simulation/control/MySimulation.hpp

```diff
@@ -668,6 +665,46 @@ class MySimulation : public SimulationHelper<simDim>
         }
     }
 
+    /** Reset all fields
+     *
+     * @param currentStep iteration number of the current step
+     */
+    void resetFields( uint32_t const currentStep )
+    {
+        auto resetField = [currentStep]( std::string const name )
+        {
+            DataConnector & dc = Environment<>::get().DataConnector();
+            auto const fieldExists = dc.hasId( name );
+            if( fieldExists )
+            {
+                using FieldHelper = SimulationFieldHelper< MappingDesc >;
+                auto field = std::dynamic_pointer_cast< FieldHelper >(
+                    dc.get< ISimulationData >( name, true )
+                );
+                if( field )
+                    field->reset( currentStep );
+                dc.releaseData( name );
+            }
+        };
+
+        /* For now the list of fields is hardcoded here, a more generic solution
```

Comment:
You could add a `@todo` here so that we can keep track of it with Doxygen.   
Otherwise, when the `DataConnector` changes we might miss this, later.

Reply:
Good point, fixed and rebased.